cli/command/container/attach.go

```diff
@@ -105,7 +105,13 @@ func RunAttach(ctx context.Context, dockerCLI command.Cli, containerID string, o
 
 	if opts.Proxy && !c.Config.Tty {
 		sigc := notifyAllSignals()
-		go ForwardAllSignals(ctx, apiClient, containerID, sigc)
+		// since the ForwardAllSignals already registers its own signal handler
+		// and should notify the daemon independently of the clients ctx cancellation
+		// we should use a background context to avoid the ForwardAllSignals from
+		// returning before all signals are forwarded to the daemon
+		bgCtx, cancel := context.WithCancel(context.Background())
```

Comment:
Instead of using `context.Background()` we can just ignore cancellation from the passed ctx:

```suggestion
		bgCtx, cancel := context.WithCancel(context.WithoutCancel(ctx))
```

This will still persist values attached to the context, so we won't lose OTEL spans for example.